src/cmd/services/m3coordinator/ingest/carbon/ingest.go

```diff
@@ -127,18 +128,21 @@ func NewIngester(
 		}
 	})
 
-	return &ingester{
+	i := &ingester{
 		downsamplerAndWriter: downsamplerAndWriter,
 		opts:                 opts,
 		logger:               opts.InstrumentOptions.Logger(),
 		tagOpts:              tagOpts,
 		metrics: newCarbonIngesterMetrics(
 			opts.InstrumentOptions.MetricsScope()),
-
-		rules: compiledRules,
-
 		lineResourcesPool: resourcePool,
-	}, nil
+		rulesUpdatedCh:    opts.RulesUpdatedCh,
+	}
+	// No need to retain watch as NamespaceWatcher.Close() will handle closing any watches
```

Comment:
Are there cases where where caller actually does manually close?

Reply:
Not currently. Just the cleanup code when the coordinator is shutting down.